store/src/java/com/zimbra/cs/mailbox/ContactBackupThread.java

```diff
@@ -0,0 +1,396 @@
+            for (Integer mailboxId : mailboxIds) {
+                if (shutdownRequested) {
+                    ZimbraLog.backup.info("%s : shutting down thread.", OPERATION);
+                    backupThread = null;
+                    return;
+                }
+                ZimbraLog.backup.debug("%s : starting to work with mailbox %d", OPERATION, mailboxId);
+                ZimbraLog.addMboxToContext(mailboxId);
+                Mailbox mbox = null;
+                try {
+                    mbox = MailboxManager.getInstance().getMailboxById(mailboxId);
+                } catch (ServiceException e) {
+                    ZimbraLog.backup.info("%s : failed to load mailbox %d, continuing to next mailbox", OPERATION, mailboxId);
+                    ZimbraLog.backup.debug(e);
+                    setContactBackuLastMailboxId(mailboxId);
+                    continue;
+                }
+                Account account = null;
+                try {
+                    account = mbox.getAccount();
+                } catch (ServiceException se) {
+                    ZimbraLog.backup.info("%s : failed to get account from mailbox %s, continuing to next mailbox", OPERATION, mbox.getAccountId());
+                    ZimbraLog.backup.debug("", se);
+                    setContactBackuLastMailboxId(mbox.getId());
+                    continue;
+                }
+                OperationContext octxt = new OperationContext(account);
+                Folder folder = getContactBackupFolder(octxt, mbox, account, true);
+                if (folder == null) {
+                    ZimbraLog.backup.info("%s : contact backup folder not found for %s, continuing to next mailbox", OPERATION, account.getName());
+                    setContactBackuLastMailboxId(mbox.getId());
+                    continue;
+                }
+                // create new backup
+                createBackup(octxt, mbox, account, sslcsf, folder, startTime);
+                // purge old backups
+                purgeOldBackups(octxt, mbox, account, folder, startTime);
+                // set current mailbox id as last processed mailbox
+            }
+            Date endTime = new Date();
+            long diff = endTime.getTime() - startTime.getTime();
+            ZimbraLog.backup.debug("%s : finished iteration on mailboxes, iteration took %d ms", OPERATION, diff);
+            long sleepThreadTime = contactBackupFreq - diff; // find actual diff so that thread will start again at a same time next time
+            ZimbraLog.backup.info("%s : Thread finished iteration, sleeping for %dms", OPERATION, sleepThreadTime);
+            sleepThread(sleepThreadTime);
+        }
+    }
+
+    private void createBackup(OperationContext octxt, Mailbox mbox, Account account, SSLConnectionSocketFactory sslcsf, Folder folder, Date startTime) {
+            HttpResponse response = exportContacts(mbox, account, sslcsf);
```

Comment:
The indent is not correct.  All the contents within the createBackup method should indent-4-space-left.